### ISSUE TITLE
Locking tracks not stopping the same train twice in short-ish intervals

### ISSUE THREAD

**Issue by jgriebler** [CovertJaguar/Railcraft#632]
When in train mode (no matter whether lockdown, boarding or holding), a locking track will not turn off once activated, even after the redstone signal has been turned off and the entire train has passed.

I have tested this with both kinds of locomotives and varying numbers and types of carts attached to them and the bug arises with all combinations I have tried.

**Comment by Forecaster** [comment]
It should turn off after some time has passed since the train cleared the track, or when a new train enters.

**Comment by jgriebler** [comment]
I haven't been able to get it to turn off by any means except by changing the mode to a non-train one. I am using a circular test track, so it is possible that it simply doesn't have enough time to turn off. On the other hand, the track doesn't turn off even if the same train reaches it again. I'll do some more testing now to see whether adding a second train or giving it more time will turn it off.

**Comment by jgriebler** [comment]
Okay, I have done more testing and it appears that you are correct: After several seconds, the track turns off like it should. It will also turn off when a different train reaches it.

However, it seems the time delay for turning off might be a bit too long; it might be preferable to be able to stop the same train again shortly after it has left the locking track. With this in mind, I would contend that the issue is not fully resolved, or at least that the behaviour of the locking track could be enhanced by simply shortening the time delay (or, if possible, removing it entirely).

**Comment by CovertJaguar** [comment]
What version of Railcraft? Also there are some tricky technical limitations to detecting trains.

**Comment by jgriebler** [comment]
I have tested it in both 9.5.0.0 and 9.6.1.0. If it's difficult to detect trains and there's no easy way to decrease the delay before turning the locking track off without breaking it, it might not be worth bothering, since the unexpected behaviour doesn't arise as often as I initially thought.

I only found the issue on a test track I built for testing an unrelated setup anyway, so I'm not sure there are really many practical examples that are broken by it.
